kernel/idle.c

```diff
@@ -59,7 +61,7 @@ void idle(void *unused1, void *unused2, void *unused3)
 		 * unmasked.  It does not take a spinlock or other
 		 * higher level construct.
 		 */
```

Comment:
This comment here  becomes a lie with this patch. :)

Reply:
The behavior is correct with the changes, what is wrong is just the reference to `k_cpu_idle`, right ? The idle hook will be called with interruptions masked, just like stated there, and will ensure that they are unmasked when returning that is the behavior that we want.